lib/util/Consts.ts

```diff
@@ -16,149 +22,68 @@ export enum TypeURL {
-  XSD_UNSIGNED_INT = 'http://www.w3.org/2001/XMLSchema#unsignedInt',
-  XSD_UNSIGNED_SHORT = 'http://www.w3.org/2001/XMLSchema#unsignedShort',
-  XSD_UNSIGNED_BYTE = 'http://www.w3.org/2001/XMLSchema#unsignedByte',
-  XSD_POSITIVE_INTEGER = 'http://www.w3.org/2001/XMLSchema#positiveInteger',
-}
-
-export const NumericTypeURLs: Set<string> = new Set(Object.values(NumericTypeURL));
-export const DerivedIntegerTypeURLs = new Set(Object.values(DerivedIntegerTypeURL));
-
-export const commonTerms: Record<string, RDF.Term> = {
-  true: DF.literal('true', DF.namedNode(TypeURL.XSD_BOOLEAN)),
-  false: DF.literal('false', DF.namedNode(TypeURL.XSD_BOOLEAN)),
-};
-
-// TODO: Rename to primitive - Should remove this and replace with aliases.
-// https://www.w3.org/TR/xmlschema-2/#built-in-primitive-datatypes
-export type Type =
-  'string'
-  | 'langString'
-  | 'date'
-  | 'boolean'
-  | 'integer'
-  | 'decimal'
-  | 'float'
-  | 'double'
-  | 'other'
-  | 'nonlexical';
-
-export type PrimitiveNumericType = 'integer' | 'decimal' | 'float' | 'double';
-export const PrimitiveNumericTypes = new Set([ 'integer', 'decimal', 'float', 'double' ]);
-
-export function type(typeURL: string): Type {
-  switch (typeURL) {
-    case null:
-    case undefined:
-    case '':
-    case TypeURL.XSD_ANY_URI: return 'string';
-    case TypeURL.XSD_STRING: return 'string';
-    case TypeURL.RDF_LANG_STRING: return 'langString';
-    case TypeURL.XSD_DATE_TIME: return 'date';
-    case TypeURL.XSD_BOOLEAN: return 'boolean';
-
-    case TypeURL.XSD_DECIMAL: return 'decimal';
-    case TypeURL.XSD_FLOAT: return 'float';
-    case TypeURL.XSD_DOUBLE: return 'double';
-    case TypeURL.XSD_INTEGER:
-    case TypeURL.XSD_NON_POSITIVE_INTEGER:
-    case TypeURL.XSD_NEGATIVE_INTEGER:
-    case TypeURL.XSD_LONG:
-    case TypeURL.XSD_INT:
-    case TypeURL.XSD_SHORT:
-    case TypeURL.XSD_BYTE:
-    case TypeURL.XSD_NON_NEGATIVE_INTEGER:
-    case TypeURL.XSD_UNSIGNED_LONG:
-    case TypeURL.XSD_UNSIGNED_INT:
-    case TypeURL.XSD_UNSIGNED_SHORT:
-    case TypeURL.XSD_UNSIGNED_BYTE:
-    case TypeURL.XSD_POSITIVE_INTEGER: return 'integer';
-    default: return 'other';
+export function type(typeString: string): LiteralTypes {
+  if (Object.values(TypeURL).includes(<TypeURL> typeString)) {
+    return <TypeURL> typeString;
   }
-}
-
-// If datatypes get lost or lose specificity during operations, we can insert a
-// concrete type, since categories should remain the same. This mostly (only)
-// relevant for integer subtypes.
-const _decategorize: Record<PrimitiveNumericType, TypeURL> = {
-  integer: TypeURL.XSD_INTEGER,
-  float: TypeURL.XSD_FLOAT,
-  double: TypeURL.XSD_DOUBLE,
-  decimal: TypeURL.XSD_DECIMAL,
-};
-
-export function decategorize(cat: PrimitiveNumericType): TypeURL {
-  return _decategorize[cat];
+  if ([ null, undefined, '' ].includes(typeString)) {
+    return TypeURL.XSD_STRING;
+  }
+  return TypeAlias.SPARQL_OTHER;
```

Comment:
We should be able to get rid of this function in it's entirety (see discussion on type-attribute). Or am I missing something?

Reply:
If by `this function` you mean `decatogorize`, it's gone?

Reply:
Something weird happened, I meant the `type(typeString: string): LiteralTypes` function.

Reply:
Yes, I was able to remove this function.